File: app/gui/MyOutputStream.js
var JsonOutputStream = require('./JsonOutputStream');

var MyOutputStream = function(output) {
    this.output = new JsonOutputStream(output);
}

MyOutputStream.prototype.write = function(type, data) {
    var jsonData = {
        type: type
    };
    if (typeof data != 'undefined') {
        for (var key in data) {
            jsonData[key] = data[key];
        }
    }
    this.output.write(jsonData);
}

module.exports = MyOutputStream;

File: app/controller/MyOutputStream.rb
require './JsonOutputStream'

class MyOutputStream
    def initialize output
        @output = JsonOutputStream.new output
    end

    def write type, data
        jsonData = {
            'type' => type
        }
        data.each do |key, value|
            jsonData[key] = value
        end
        @output.write jsonData
    end
end
